lib/ngx/re.lua

```diff
@@ -121,24 +130,21 @@ function _M.split(subj, regex, opts, ctx, max, res)
         return res
     end
 
-    if regex == "" then
-        local pos = ctx.pos
-        local last = len
-        if max > 0 then
-            last = math_min(len, pos + max - 1)
-        end
+    local res_idx = 0
+    local sub_idx = ctx.pos
 
-        local res_idx = 1
-        while pos < last do
-            res[res_idx] = sub(subj, pos, pos)
-            res_idx = res_idx + 1
-            pos = pos + 1
-        end
+    local start_regex = find(regex, "^", nil, true) ~= nil
```

Comment:
> Maybe you should try the "m" regex flag in the Lua split?

You mean require users to add it themselves in the opts argument? Because Perl's `split` behavior is - apparently - to be that it adds the `m` automatically when `/^/` is given.